### PR TITLE
fix(hydrated_cubit): support use alongside hive

### PR DESCRIPTION
**Pull request by orsenkucher**
Basically when Hive is running alongside HydratedBloc, latest invocation of `Hive.init()` (which can occur inside HB, or somewhere in main) overrides previous Hive configuration.

We address this issue by using our own `HiveImpl` instance, which can live inside or outside Hive's default directory

### CODE REVIEW

**Comment by orsenkucher** [comment]
@semantic-pull-requests what do you want from me bro😭?

File: packages/hydrated_cubit/lib/src/hydrated_storage.dart [hunk 1]
```diff
@@ -52,15 +54,20 @@ class HydratedStorage implements Storage {
     return _lock.synchronized(() async {
       if (_instance != null) return _instance;
       final directory = storageDirectory ?? await getTemporaryDirectory();
-      if (!kIsWeb) Hive.init(directory.path);
-      final box = await Hive.openBox<dynamic>(
+      hive = HiveImpl();
```
Comment (felangel) [inline packages/hydrated_cubit/lib/src/hydrated_storage.dart:59]: Can you please provide some context for why this change is needed? Thanks 👍

Comment (felangel) [inline packages/hydrated_cubit/lib/src/hydrated_storage.dart:59]:
```suggestion
      // Need to use HiveImpl directly to avoid conflicts with existing Hive initialization
      // https://github.com/hivedb/hive/issues/336
      hive = HiveImpl();
```

File: packages/hydrated_cubit/lib/src/hydrated_storage.dart [hunk 2]
```diff
@@ -52,15 +54,20 @@ class HydratedStorage implements Storage {
...
+      if (!kIsWeb) hive.init(directory.path);
+      final box = await hive.openBox<dynamic>(
         'hydrated_box',
         encryptionCipher: encryptionCipher,
       );
       return _instance = HydratedStorage(box);
     });
   }
 
+  /// Own instance of HiveImpl, to resolve interference
```
Comment (felangel) [inline packages/hydrated_cubit/lib/src/hydrated_storage.dart]:
```suggestion
  /// Internal instance of [HiveImpl].
  /// It should only be used for testing.
```

File: packages/hydrated_cubit/test/hydrated_storage_test.dart [hunk 3]
```diff
@@ -61,9 +61,9 @@ void main() {
         expect(instanceA, instanceB);
       });
 
-      test('calls Hive.init with correct directory', () async {
+      test('creates own HiveImpl with correct directory', () async {
```
Comment (felangel) [inline packages/hydrated_cubit/test/hydrated_storage_test.dart]:
```suggestion
      test('creates internal HiveImpl with correct directory', () async {
```

**Review by felangel — DISMISSED** [review]
LGTM :shipit:

File: packages/hydrated_cubit/lib/src/hydrated_storage.dart [hunk 4]
```diff
@@ -52,15 +54,23 @@ class HydratedStorage implements Storage {
     return _lock.synchronized(() async {
       if (_instance != null) return _instance;
       final directory = storageDirectory ?? await getTemporaryDirectory();
-      if (!kIsWeb) Hive.init(directory.path);
-      final box = await Hive.openBox<dynamic>(
+      // Need to use HiveImpl directly to avoid conflicts with existing Hive initialization
```
Comment (felangel) [inline packages/hydrated_cubit/lib/src/hydrated_storage.dart]:
```suggestion
      // Use HiveImpl directly to avoid conflicts with existing Hive.init
```
